### PR TITLE
FPC bug fixes for i386-win32

### PR DESCRIPTION
**Pull request by micha137**
* by @CuriousKit
* formerly PR #123

### CODE REVIEW

File: Source/GR32_System.pas [hunk 1]
```diff
@@ -226,7 +226,7 @@ function GetProcessorCount: Cardinal;
     ($800000,  $400000, $2000000, $4000000, $80000000, $40000000);
     {ciMMX  ,  ciEMMX,  ciSSE   , ciSSE2  , ci3DNow ,  ci3DNowExt}
 
-function CPUID_Available: Boolean;
+function CPUID_Available: Boolean; assembler;
```
Comment (andersmelander) [inline Source/GR32_System.pas]: Needs {$IFDEF FPC}...{$ENDIF}

Comment (CuriousKit) [reply Source/GR32_System.pas]: If I'm right in thinking, Delphi doesn't actually care if 'assembler' is there or not, but will do.

**Comment by CuriousKit** [comment]
> 
> 
> Pls add missing {$IFDEF FPC}...{$ENDIF}

[i386_FPC_asm_fixes.zip](https://github.com/graphics32/graphics32/files/4140297/i386_FPC_asm_fixes.zip)
Missing $IFDEFs added.

**Comment by CuriousKit** [comment]
This is the patch with just the IFDEF fixes.  I can't push it to the branch myself because I don't have permission.

[IFDEFFix.zip](https://github.com/graphics32/graphics32/files/4140355/IFDEFFix.zip)

**Comment by CuriousKit** [comment]
Is the fixed patch okay?

**Comment by micha137** [comment]
@andersmelander can i merge this now?

**Comment by andersmelander** [comment]
> can i merge this now?

@micha137 Sure. I've already approved the pull request.

**Comment by micha137** [comment]
> > can i merge this now?
> 
> @micha137 Sure. I've already approved the pull request.

I'm asking because I wondered why you didn't merge it yourself yet.

**Comment by andersmelander** [comment]
> I'm asking because I wondered why you didn't merge it yourself yet.

My approval basically just meant "looks good to me".
I didn't want to merge it myself because I don't have FPC installed and haven't verified that it compiles with Delphi.

**Comment by CuriousKit** [comment]
I'm glad you approve.  Thank you everyone.  I hope we can work again!

**Comment by micha137** [comment]
> I didn't want to merge it myself because I don't have FPC installed and haven't verified that it compiles with Delphi.

My Jenkins says that it compiled in Delphi.  I also don't have FPC installed :)

**Pull request merged**